samples/python/src/common/base_server_executor.py

```diff
@@ -70,8 +68,7 @@ def __init__(
       self._supported_extension_uris = {ext.uri for ext in supported_extensions}
     else:
       self._supported_extension_uris = set()
-    system_utils.check_google_api_key()
-    self._client = genai.Client(api_key=os.getenv("GOOGLE_API_KEY"))
```

Comment:
This can probably be kept mostly as is, just get the environment variables and load them into the client.